### ISSUE TITLE
Namespacing doesn't not work fully for multi-page builds

### ISSUE THREAD

**Issue by sholladay** [jrburke/r.js#851]
Whenever I use the `modules` option in combination with `namespace`, it results in calls to `require()` not being namespaced within the built files, whereas the calls to `define()` _are_. Switching the config to use `name` instead of `modules`, the exact same files have the expected behavior, both `require()` and `define()` get namespaced.

This is easy to reproduce on my end.

**Comment by sholladay** [comment]
@jrburke sorry to be a pest, but this is making it difficult for me to release an app ... any quick pointers would be appreciated

**Comment by jrburke** [comment]
Sorry I have not looked closer yet, some quick thoughts:

1) Make sure [skipDirOptimize](https://github.com/jrburke/r.js/blob/2fdfddd959f371b775027c0145b00aed334a58f6/build/example.build.js#L111) is **not** set to true.

2) If not minifying (by saying optimize: none), then you may want to set [normalizeDirDefines](https://github.com/jrburke/r.js/blob/2fdfddd959f371b775027c0145b00aed334a58f6/build/example.build.js#L132) to "all", or just try "all" to see if that helps.

**Comment by sholladay** [comment]
Hmm. Well that did help collect more data.

Here is the behavior I am seeing:

1. If I set `normalizeDirDefines: 'all'`, it fixes the issue in some places, but not in others.
2. I am able to see a pattern in the "some places" from number 1. It fixes it in places where the first item in the deps array argument to `require` is a string literal. Anything else still fails, such as if it is a variable or if the array is empty or not supplied at all, etc.

**Comment by sholladay** [comment]
FYI, I have placed a [$100 bounty](https://www.bountysource.com/issues/26479387-incomplete-namespacing-in-dir-output) on this issue.

**Comment by PlasmaPower** [comment]
@sholladay Sorry I'm new to this project but could you send me a zip reproducing the issue along with a quick explanation of it should do?

**Comment by sholladay** [comment]
Yes, I will try to do that within the next few days. This bug is from over half a year ago, I'll have to brush up on it. At the time, I had to just hack around it.

**Comment by PlasmaPower** [comment]
@sholladay Makes sense. Thanks for the help!
